lib/context-menu-handlers.js

```diff
@@ -33,16 +32,24 @@ self.on('click', function (node, data) {
   self.postMessage({
     src: node.href || node.src,
     url: window.location.href,
-    domain: host // new URL(node.href || node.src).host
+    domain: host
   });
 });`;
 
 const contextMenuSoundcloudContentScript = `
-self.on('click', function (node) {
-  self.postMessage({
-    url: node.href,
-    domain: 'soundcloud.com'
-  });
+self.on('click', function (node, data) {
+  // if the target element isn't the link we
+  // are looking for, recurse until we find it.
+  function setParentNode(node) {
+    if (node.href) {
```

Comment:
The functin name is kinda funny, now that I'm looking at it. Might be nicer as a while loop. Just a suggestion, though. Fine to leave as-is.